### PR TITLE
Fix custom pass reorder issue

### PR DESCRIPTION
**Pull request by alelievr**
### Purpose of this PR
Fix an error in the console when re-ordering custom passes: https://fogbugz.unity3d.com/f/cases/1290618/

Context:  
Since the 2021.1.0a5, a fix for the SerializeReference attribute has landed and broken the custom pass re-ordering. This fix will be backported on 2020.2, 2020.1, and 19.4 but the issue in HDRP only happens in 11.x, 10.x, and 9.x.

---
### Testing status
Tested that reordering and removing custom passes does not generate any errors. Did not see any other issues

---
### Comments to reviewers
Investigation notes with Martin: https://unity.slack.com/archives/C9SQHJGN6/p1605179395179800

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeEditor.cs [hunk 1]
```diff
@@ -235,6 +235,21 @@ void CreateReorderableList(SerializedProperty passList)
...
+                ReorderableList.defaultBehaviours.DoRemoveButton(list);
+                ClearCustomPassCache();
+            };
+        }
+
+        void ClearCustomPassCache()
+        {
+            // When custom passes are re-ordered, a topological change happens in the SerializedProperties
+            // So we have to rebuild all the drawers that were storing SerializedProperties.
+            // customPassDrawers.Clear();
```
Comment (martin-unity3d) [inline com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeEditor.cs:252]: Shouldn't this line be uncommented? Thank you!

Comment (alelievr) [reply com.unity.render-pipelines.high-definition/Editor/RenderPipeline/CustomPass/CustomPassVolumeEditor.cs]: Yes! I'm not sure what happened there ^^'
Thanks!

**Review by remi-chapelain — APPROVED** [review]
Made sure that not backporting this fix to 7.x/8.x won't introduced more regression in 19.4 ✔️ 
Testing seems sufficient ! ✔️

**Review by sebastienlagarde — APPROVED** [review]

**Comment by sebastienlagarde** [comment]
2020.2: https://ono.unity3d.com/unity/unity/pull-request/115733/_/2020.2/serialization/serializeref_reuseinstances#
2020.1: https://ono.unity3d.com/unity/unity/pull-request/115858/_/2020.1/serialization/serializeref_reuseinstances

**Pull request merged**